app/services/authenticator.ts

```diff
@@ -72,20 +72,30 @@ export default class AuthenticatorService extends Service {
     await this.syncCurrentUser();
   }
 
-  initiateLogin(redirectPath: string | null) {
-    const isFullUrl = redirectPath?.startsWith(`${config.x.backendUrl}`);
-
-    if (isFullUrl) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + redirectPath;
-    } else if (redirectPath) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${redirectPath}`);
-    } else if (this.router.currentURL) {
+  initiateLogin(redirectPathOrUrl: string | null = null) {
```

Comment:
```suggestion
  initiateLogin() {
```

Let's remove the optional parameter here - let callers call `initiateLoginAndRedirectTo()` if needed. 

The intention here is to simplify the interface, and not have too many branches that the caller needs to reason about - by adding this branch we haven't really made this any simpler.

Reply:
I was thinking exposing a single function was better, so we could internally handle all the complexity. And not make callers decide which one to call.

Reply:
Hmm, then why have two functions at all? Just keep one (and make the other private?)?

That's fine by me too, would be a simple enough interface

Reply:
@ryan-gang one way to think about this is "intention" - 90% of our cases, the intention is just to initiate login with the current URL as the redirect target. It's only in the other 10% where we need to redirect a user to a specific location. Given that split, keeping two functions could make sense.